src/gcp/cloudscheduler.ts

```diff
@@ -159,17 +167,28 @@ export async function createOrReplaceJob(job: Job): Promise<any> {
 
 /**
  * Check if two jobs are functionally equivalent.
- * @param job a job to compare.
- * @param otherJob a job to compare.
+ * @param existingJob a job to compare.
+ * @param newJob a job to compare.
  */
-function isIdentical(job: Job, otherJob: Job): boolean {
-  return (
-    job &&
-    otherJob &&
-    job.schedule === otherJob.schedule &&
-    job.timeZone === otherJob.timeZone &&
-    _.isEqual(job.retryConfig, otherJob.retryConfig)
-  );
+function needUpdate(existingJob: Job, newJob: Job): boolean {
+  if (!existingJob) {
```

Comment:
was it intentional that we're now only checking whether one of the parameters is truthy?

Reply:
I thought it was redundant! But since it's safer to check for newJob too, I'll reinsert the logic.